src/helpers/pivot/pivot_domain_helpers.ts

```diff
@@ -253,3 +253,27 @@ export function getRunningTotalDomainKey(
   }
   return domainToString([...domain.slice(0, index), ...domain.slice(index + 1)]);
 }
+
+export function sortPivotTree(
+  tree: DimensionTree,
+  baseDomain: PivotDomain,
+  sortFn: (a: PivotDomain, b: PivotDomain) => number
+) {
+  const sortedTree = [...tree];
+  const domain = [...baseDomain];
+  sortedTree.sort((node1, node2) =>
+    sortFn(
+      [...domain, { field: node1.field, value: node1.value, type: node1.type }],
+      [...domain, { field: node2.field, value: node2.value, type: node2.type }]
```

Comment:
nitpick, we can avoid creating new objects as both types are compatible
```suggestion
      [...domain, node1],
      [...domain, node2]
```